### PR TITLE
Update gene sets and denovo gene sets interface 

### LINKED ISSUE #81: Update gene sets and denovo gene sets interface

### PR DESCRIPTION
**Pull request by iordanivanov**
closes #81

### CODE REVIEW

File: dae/dae/gene/denovo_gene_sets_db.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            permitted_datasets
+        )
+
+        return DenovoGeneSet.get_gene_set(
+            gene_set_id,
+            list(self._denovo_gene_set_cache.values()),
+            denovo_gene_sets_types
+        )
+
+    def get_gene_sets(self, denovo_gene_set_spec, permitted_datasets=None):
```
[the remark this answers is not in the document]
Comment (lchorbadjiev) [reply dae/dae/gene/denovo_gene_sets_db.py]: probably we should rename this to 
get_all_gene_sets(denovo_gene_set_spec, permitted_genotype_data=None)

File: dae/dae/gene/gene_sets_db.py [hunk 2]
```diff
@@ -17,30 +17,31 @@ def __init__(self, gene_sets_collection_id, config):
...
                     return genes[x].sym
-            gene_sets_collection.renameGenes("sym", rF)
+            gene_sets_collection.renameGenes('sym', rF)
 
-        if gene_sets_collection.geneNS != 'sym':
-            raise Exception('Only work with id or sym namespace')
         return gene_sets_collection
 
-    def load(self):
+    def load(self, from_file=False):
```
[the remark this answers is not in the document]
Comment (lchorbadjiev) [reply dae/dae/gene/gene_sets_db.py]: `from_file` should be removed... here we read what is defined in the configuration

File: dae/dae/gene/gene_sets_db.py [hunk 3]
```diff
@@ -52,15 +53,12 @@ def load(self):
         ]
         return self.gene_sets_collections
 
-    def get_gene_sets(self, gene_sets_types=[], **kwargs):
+    def get_gene_sets(self):
```
Comment (lchorbadjiev) [inline dae/dae/gene/gene_sets_db.py]: rename to be similar to denovo gene sets db

File: dae/dae/gene/gene_sets_db.py [hunk 4]
```diff
@@ -118,7 +121,15 @@ def has_gene_sets_collection(self, gsc_id):
             for gsc in self.get_collections_descriptions()
         ])
 
-    def _load_gene_sets_collection(self, gene_sets_collection_id, load=True):
+    @staticmethod
+    def load_gene_set_from_file(filename, config):
```
Comment (lchorbadjiev) [inline dae/dae/gene/gene_sets_db.py]: this should load a single gene set outside of all collections...

File: dae/dae/gene/gene_sets_db.py [hunk 5]
```diff
@@ -17,30 +17,31 @@ def __init__(self, gene_sets_collection_id, config):
...
 
-    def _load(self):
-        try:
-            gene_sets_collection = \
-                GeneInfoConfigParser.getGeneTerms(self.config, self.gsc_id)
-        except Exception:
-            traceback.print_exc()
-            gene_sets_collection = loadGeneTerm(self.gsc_id)
+    @staticmethod
+    def _load_collection(config, gene_sets_collection):
```
Comment (lchorbadjiev) [inline dae/dae/gene/gene_sets_db.py]: this should be renamed and reorganized to match what method really do...

Comment (lchorbadjiev) [reply dae/dae/gene/gene_sets_db.py]: probably the constructor should accept GeneInfo instance and use that instance when needed...

File: dae/dae/gene/denovo_gene_sets_db.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+        denovo_gene_set = DenovoGeneSet(study, denovo_gene_set_config)
+        denovo_gene_set.load()
+
+        self._denovo_gene_set_config_cache_values[denovo_gene_set_id] = \
+            denovo_gene_set_config
+        self._denovo_gene_set_cache_values[denovo_gene_set_id] = \
+            denovo_gene_set
+
+    def _build_cache(self, gene_set_ids=None):
```
[the remark this answers is not in the document]
Comment (iordanivanov) [reply dae/dae/gene/denovo_gene_sets_db.py]: I've changed the method to load and build only the supplied gene set ids - please confirm if this is correct.

[the remark this answers is not in the document]
Comment (iordanivanov) [reply dae/dae/gene/denovo_gene_sets_db.py]: the method now simply builds and saves the denovo gene set collections for the supplied genotype data ids

**Pull request merged**